pandas/util/testing.py

```diff
@@ -515,9 +516,14 @@ def equalContents(arr1, arr2):
     return frozenset(arr1) == frozenset(arr2)
 
 
-def assert_index_equal(left, right, exact='equiv', check_names=True,
-                       check_less_precise=False, check_exact=True,
-                       check_categorical=True, obj='Index'):
+def assert_index_equal(left: Index,
+                       right: Index,
```

Comment:
Instead of a raw Index, is it possible to add a typing.TypeVar to _typing.py:

```python
AnyIndex = TypeVar('AnyIndex ', pd.Index, pd.RangeIndex, pd.Int64Index, pd.MultiIndex, + other index types...)
```

and then use that here (I'm not super aqianted with typing, so could be wrong)?

That would avoid the need to adapt the code below.

Reply:
this is ok as all of these are Index subclasses

Reply:
The point was that AnyIndex would capture .levels and other attributes that are not set on the base Index.

Reply:
I think this is OK too. MyPy would still complain about those attributes not existing on Index